pkgs/misc/drivers/argononed/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, stdenv, fetchFromGitLab, dtc, installShellFiles }:
+
+stdenv.mkDerivation rec {
+  pname = "argononed";
+  version = "unstable-2022-03-26";
+
+  src = fetchFromGitLab {
+    owner = "DarkElvenAngel";
+    repo = pname;
+    rev = "97c4fa07fc2c09ffc3bd86e0f6319d50fa639578";
+    sha256 = "sha256-5/xUYbprRiwD+FN8V2cUpHxnTbBkEsFG2wfsEXrCrgQ=";
+  };
+
+  postPatch = ''
+    patchShebangs configure
+  '';
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  buildInputs = [ dtc ];
+
+  installPhase = ''
```

Comment:
Instead of overwriting installPhase (which would require you to call hooks). You could use `postInstall` instead.

Reply:
The missing hooks are:
`runHook preInstall`
`runHook postInstall`

Reply:
Better not abuse `postInstall` phase for that. Otherwise people trying to overwrite the package will have some surprises.

Reply:
Anywya, good catch. Please add the pre/post hooks.